components/BottomNavigation/tests/unit/beta/MDCBottomNavigationBarControllerTests.m

```diff
@@ -0,0 +1,348 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MDCBottomNavigationBarController.h"
+
+static CGFloat const kDefaultExpectationTimeout = 15;
```

Comment:
Please import CoreGraphics to use CGFloat.

Reply:
Can do.  Just curious, why the redundant import if CoreGraphics is a dependency of UIKit?

Is it if UIKit ever changes their dependency on CoreGraphics this code won't break?

Reply:
Importing direct dependencies is a habit I've picked up over the years and think it's a good practice in general.  

For example, if I import "MaterialButtons+Theming.h", it could import "MaterialButtons.h" for me. However in the future that might change to a forward declaration (it won't), and then my code would break.  If I explicitly import all of my direct dependencies, my code remains buildable.

Also, as a library developer I've experienced the pain of having to fix many clients' code because they depended on transitive imports for their direct dependencies. :-\

Reply:
As an aside, I don't see any imports for CoreGraphics in UIKit.h. Here's what my SDK headers have:

* `Foundation`
* `UIKitCore`
* A conditional import of `UIKit/DocumentManager`

Reply:
Makes sense!

UIGraphics, UIBezierPath, and others import CG.  Those are both imported by UIKitCore.